FEAT: Add User Info Headers to External Rerank Model Retrieval

rgaricano opened the pull request:
## Add User Info Headers to External Rerank Model Retrieval

Feature Petition Discussion : https://github.com/open-webui/open-webui/issues/15572#issuecomment-3048584786
-------------


----


### Description

- Added User Info Headers to External Rerank Model Retrieval.
-Added user parameter to ExternalReranker call.


### Contributor License Agreement

By submitting this pull request, I confirm that I have read and fully agree to the [Contributor License Agreement (CLA)](/CONTRIBUTOR_LICENSE_AGREEMENT), and I am providing my contributions under its terms.

rgaricano committed:
FEAT: Add User Info Headers to External Rerank Model Retrieval

## Add User Info Headers to External Rerank Model Retrieval

Feature Petition Discussion : https://github.com/open-webui/open-webui/issues/15572#issuecomment-3048584786
-------------
----

rgaricano committed:
FEAT: Add User Info Headers to External Rerank Model Retrieval

## Add User Info Headers to External Rerank Model Retrieval

Feature Petition Discussion : https://github.com/open-webui/open-webui/issues/15572#issuecomment-3048584786
-------------

----

### Description

- Added User Info Headers to External Rerank Model Retrieval.

### Contributor License Agreement

By submitting this pull request, I confirm that I have read and fully agree to the [Contributor License Agreement (CLA)](/CONTRIBUTOR_LICENSE_AGREEMENT), and I am providing my contributions under its terms.

rgaricano committed:
Merge branch 'open-webui:dev' into dev

rgaricano committed:
Update retrieval.py - ensure that user exists and is verified

ensure that user exists and is verified

rgaricano commented on backend/open_webui/routers/retrieval.py:
```diff
@@ -167,6 +167,7 @@ def get_rf(
                         url=external_reranker_url,
                         api_key=external_reranker_api_key,
                         model=reranking_model,
+                        user=user,
```
added  user=Depends(get_verified_user)
to ensure that user exists and is verified

athoik replied:
Great! Please check also my other comment about headers, using same logic as in other places is preffered.
Also the other logic checks if user is not None.

rgaricano committed:
Update external.py - user 

Correction to adapt change in retrieval that pass verify user object, and adapt it to code use schema.

rgaricano committed:
Update external.py - adapt

correction to keep same code format as in other places.

rgaricano committed:
Update external.py remove innecesarry spaces

rgaricano commented on backend/open_webui/retrieval/models/external.py:
```diff
@@ -36,12 +37,19 @@ def predict(self, sentences: List[Tuple[str, str]]) -> Optional[List[float]]:
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {self.api_key}",
+            }
+            if ENABLE_FORWARD_USER_INFO_HEADERS:
```
I saw it, I thought about doing it this way, but I'm not very comfortable with that notation and I was in a bit of a hurry.
I'll take note.
Also I prefer have the call clear and do the asignation before, but I keep as is used in other places.

tjbck commented on backend/open_webui/retrieval/models/external.py:
```diff
@@ -41,6 +42,15 @@ def predict(self, sentences: List[Tuple[str, str]]) -> Optional[List[float]]:
+                            "X-OpenWebUI-User-Email": self.user.email,
+                            "X-OpenWebUI-User-Role": self.user.role,
+                        }
+                        if ENABLE_FORWARD_USER_INFO_HEADERS and user
```
`self.user`

athoik commented:
Supersede by #15615
